core/src/main/java/hudson/model/User.java

```diff
@@ -973,6 +975,19 @@ public Object getDynamic(String token) {
     public ContextMenu doContextMenu(StaplerRequest request, StaplerResponse response) throws Exception {
         return new ContextMenu().from(this,request,response);
     }
+    
+    /**
+     * Gets list of Illegal usernames, for which users should not be created.
+     * Always includes users from {@link #ILLEGAL_PERSISTED_USERNAMES}
+     * @return List of usernames
+     */
+    @Restricted(NoExternalUse.class)
+    /*package*/ static Set<String> getIllegalPersistedUsernames() {
```

Comment:
Shouldn't you be using that method above in the foreach too?

NIT: maybe a case for @VisibleForTesting;